source/extensions/filters/http/composite/action.h

```diff
@@ -39,8 +40,31 @@ class ExecuteFilterActionFactory
             composite_action.typed_config());
     ProtobufTypes::MessagePtr message = Config::Utility::translateAnyToFactoryConfig(
         composite_action.typed_config().typed_config(), validation_visitor, factory);
-    auto callback = factory.createFilterFactoryFromProto(*message, context.stat_prefix_,
-                                                         context.factory_context_);
+
+    Envoy::Http::FilterFactoryCb callback = nullptr;
+
+    // TODO(tyxia) Update the logic later to create the filter from the `factoryContext` first if it
+    // is present.
+    try {
```

Comment:
I think this should be TRY_ASSERT_MAIN_THREAD?  

unfortunately our checks in check_format completely exclude extensions, and I think we should require all new code assert main thread, so I'm going to extend the format checks to disallow raw try for extensions and switch this (and other points) to TRY_NEEDS_AUDIT.  If you can confirm and update this one that'd be lovely.

Reply:
Sorry, Alyssa. I missed this comment somehow. I think ASERT_MAIN_THREAD make sense.

I am following up with a change to address the TODO I have above, which will eliminate the needs of {try, catch} here